### PR TITLE
Migration of semileptonic trigger SF library to hist

### PR DESCRIPTION
**Pull request by mmarchegiani**
The present PR re-implements the semileptonic trigger SF library in order to migrate from `Coffea` histograms to `hist` histograms.

### CODE REVIEW

File: PocketCoffea/utils/PlotEfficiency.py [hunk 1]
```diff
@@ -6,8 +6,9 @@
 
 import mplhep as hep
 import hist
+from hist import Hist
 from coffea import processor
-from coffea.hist import Bin, Hist, plot
+#from coffea.hist import Bin, Hist, plot
```
Comment (valsdav) [inline PocketCoffea/utils/PlotEfficiency.py:11]: Let's delete completely the outdated lib

File: PocketCoffea/workflows/base.py [hunk 2]
```diff
@@ -116,7 +109,7 @@ def load_metadata(self):
         self._year = self.events.metadata["year"]
         self._triggers = triggers[self.cfg.finalstate][self._year]
         self._btag = btag[self._year]
-        self._isMC = self.events.metadata["isMC"]
+        self._isMC = self.events.metadata["isMC"] == "True"
```
Comment (valsdav) [inline PocketCoffea/workflows/base.py:112]: Is this a string at the end in the dataset definition?

Comment (valsdav) [reply PocketCoffea/workflows/base.py:112]: Probably better to be sure in build_dataset to have a boolean..

**Review by valsdav — COMMENTED** [review]
The changes are fine for me!

File: PocketCoffea/workflows/base.py (hunk 2 above)
Comment (mmarchegiani) [reply PocketCoffea/workflows/base.py:112]: Yes, at the moment the attribute `isMC` is a string. This needs to be fixed when building the json dataset.

**Pull request merged**